### ISSUE TITLE
Make clearer for learners how to increase the text size on the app.

### ISSUE THREAD

**Issue by seanlip** [oppia/design-team#159]
**Platform**
Oppia Android

**Describe the request**

A couple of users have asked us to:

- Make the text a bit bigger for readability (1 user on Play Store)
- "I couldn't make the text bigger, which made reading a bit difficult. Adjustable text size would be better." (1 user, Mar 2024)

However, we actually have this functionality already in the lesson player! The feedback above suggests that it might not be discoverable enough.

Some things to consider:
- The current functionality only affects the lesson player, not other things in the app. Is that an issue? Should we rethink the feature and make it profile-wide?
- Should a user be updating the text size in the phone system settings instead? If so, how do we train them to do that?

**Context for the request**

This is a suggested improvement to the current Android app.

**UXR success criteria**

- Users should be able to discover how to change the text size, and successfully do so.

**DESIGN APPROVAL CHECKLIST -- DO NOT EDIT.**

- [ ] UX Design approved (Android mocks should include dark mode)
- [ ] UX Writing approved
- [ ] Design Systems Team approved
- [ ] Product approved

_For a design project to be completed, it needs approval from UXD, UXW, DS and PM. Please tag the following reviewers when your project is ready (at least one reviewer for each of the 4 bullet points below):_
- UXD: 
- UXW: @ccc253 / @tamiorendain
- Design systems: 
- PM: @seanlip

**Comment by seanlip** [comment]
Initial mocks: https://www.figma.com/design/pREu0cu2YojRV0jiU9YRba/Text-Size-Adjuster-Oppia?node-id=0-1&p=f&t=viHhTRsDkIj4gFwb-0 (thanks @HayunChelseaKim!)

**Comment by tamiorendain** [comment]
Adding this to our queue. Seems like our only assignment is to review the "text size adjuster" copy/screen

**Comment by einstein395** [comment]
Taking over the assignment. Can I please get edit access to the Figma file? 
@tamiorendain  are we only updating the copy "Text size adjuster"?

Thanks, 
Aparna

**Comment by tamiorendain** [comment]
@einstein395 I believe it's the "text size adjuster" screen - and also maybe the kebab screen? @seanlip can you confirm?

A note for you @einstein395 - unless otherwise noted, feel free to add your edits/suggestions in Figma as comments :)

**Comment by einstein395** [comment]
Sounds good Tami, I will add my comments in Figma.

**Comment by seanlip** [comment]
> [@einstein395](https://github.com/einstein395) I believe it's the "text size adjuster" screen - and also maybe the kebab screen? [@seanlip](https://github.com/seanlip) can you confirm?

Yup, anything in Figma (except the lesson text) is fair game for commenting on. Thanks!

**Comment by einstein395** [comment]
Thanks Sean, I will get started on this and will leave my comments in the Figma

**Comment by einstein395** [comment]
Hi @seanlip @tamiorendain, I have made some suggestions for the text and design in the Figma File. Please let me know if you have any questions.

**Comment by seanlip** [comment]
Thanks @einstein395! Just a note, @HayunChelseaKim is the designer working on this project, so I suggest working directly with her (as PM, I generally come in just for the final review stage). 

Thanks!

**Comment by einstein395** [comment]
Thanks Sean, i will reach out to Chelsea. @HayunChelseaKim, can you please see my comments in the Figma file. Please let me know if you have any questions.

**Comment by einstein395** [comment]
Hi Sean, 

Can you please let me know who the UX designer is on this task? Is Chelsea still the designer on this project?

Thanks, 
Aparna

**Comment by seanlip** [comment]
Hi @einstein395, I received a note from @HayunChelseaKim yesterday that, unfortunately, she cannot continue on the team. So this project is temporarily unassigned, sorry about that! 

I'm also leaving some notes here for handover to the next designer. Here is a screenshot of the [Figma file](https://www.figma.com/design/pREu0cu2YojRV0jiU9YRba/Text-Size-Adjuster-Oppia?node-id=0-1&p=f&t=7tLu5yCrLKOQgLfN-0) and a synopsis of the relevant comments:

![Image](https://github.com/user-attachments/assets/77ac813f-086e-47bf-a855-2ed24c5485d1)

------

Notes/Comments:

- Currently, the top right "kebab button" (the one with 3 dots), when clicked, brings up the bottomsheet shown [here](https://drive.google.com/file/d/1WXPkZTDdAb4mBxSpXhIwXMA02qGp457M/view?usp=drive_link). That's where it says "Options".

- Comments from the Android tech lead (which suggest that we might need to consider a different approach):
    - It seems that the slider in the mocks is designed in this way to provide a "real-time" feedback on text size changing in the background. There's one fairly significant technical limitation that will make this impossible currently: we have to recreate the whole activity to apply text changes. This means that changing the slider will be result in a "flashing" effect with the whole screen being recreated, and then the slider pop-up reopening. I think we can still achieve what's desired with some cleverness, but it can't be the entire screen without that flashing effect (without what I would suspect is substantial changes). We could still navigate the user to a new screen with a slider and demonstrate a preview with text that could be changed in real-time. However, we'd also need an "apply changes" button to actually recreate the whole activity (or make that change upon navigating away from the screen).
    - Separately, a slider is sort of a strange control for this unless we explicitly want to support variable font sizes (which we don't today--they are presets). I'm not convinced on variability, it's easy to get a bit frustrated picking the "correct" zoom. If we're sure we want a slider, though, perhaps we could use snapping and ticks in the slider to make it easier to select different choices (see discrete sliders in https://m2.material.io/components/sliders#specs).-     
 
- Comments from the Android dev team (they don't need to be followed directly, but might be a good starting point for other versions to test / changes to consider):
    -  The toolbar already has very squeezed real estate so are we open  to having the textsize icon disappear into the kebab menu as it is today, but instead of needing to go to options screen first, we will add textsize as one of the bottomsheet menu's options? Another benefit to this is that, when the kebab icon is clicked, the options menu dialog will appear, showing the icon along with the associated name, so they can figure out what the icon means.
    - Is there a specific reason for naming it "Options"? "Settings" might be better. Additionally, the (possibly outdated) [M1 Material Design pattern guidelines](https://m1.material.io/patterns/settings.html#:~:text=All%20of%20an%20app%E2%80%99s%20settings%20should%20be%20reached%20via%20the%20%E2%80%9CSettings%E2%80%9D%20label.%20Do%20not%20use%20synonyms%20such%20as%20%E2%80%9COptions%E2%80%9D%20or%20%E2%80%9CPreferences.%E2%80%9D) mention that "All of an app’s settings should be reached via the 'Settings' label. Do not use synonyms such as 'Options' or 'Preferences.'"

**Comment by chloevillanu** [comment]
Hello! I created a new Figma file since the previous one was locked to the UX designer who was originally assigned to this:
https://www.figma.com/design/lK6lTSgh6K9iphIfuXkOLT/Text-Size-Adjuster-Oppia-%23159?node-id=0-1&t=qQUfoUcq8f4cyygH-1

- I have just joined and getting used to navigating the notes/comments from the previous designer, design system, etc. I’d appreciate any feedback on any mistakes I might have made to help me get up to speed faster. :)
- Additionally, I haven’t been able to get my hands on an Android device yet—I’ll have access later this week—so I’m relying on the notes provided.

**Comment by seanlip** [comment]
Adding some notes here from @BenHenning (Android tech lead):

- To add to what's already been discussed: it seems that the slider in the mocks is designed in this way to provide a "real-time" feedback on text size changing in the background. There's one fairly significant technical limitation that will make this impossible currently: we have to recreate the whole activity to apply text changes. This means that changing the slider will be result in a "flashing" effect with the whole screen being recreated, and then the slider pop-up reopening. I think we can still achieve what's desired with some cleverness, but it can't be the entire screen without that flashing effect (without what I would suspect is substantial changes). We could still navigate the user to a new screen with a slider and demonstrate a preview with text that could be changed in real-time. However, we'd also need an "apply changes" button to actually recreate the whole activity (or make that change upon navigating away from the screen).
- Separately, a slider is sort of a strange control for this unless we explicitly want to support variable font sizes (which we don't today--they are presets). I'm not convinced on variability, it's easy to get a bit frustrated picking the "correct" zoom. If we're sure we want a slider, though, perhaps we could use snapping and ticks in the slider to make it easier to select different choices (see discrete sliders in https://m2.material.io/components/sliders#specs).
- If the slider has discrete ticks, that will help solve the "flashing" issue. An alternative might be to have a non-discrete slider with a preview that resizes in realtime (this is both possible and something I've seen other apps do). We'd probably want to add some sort of "apply" button to actually 'save' the setting (and reload the app to use it).
